doc/man7/property.pod

```diff
@@ -0,0 +1,200 @@
+=pod
+
+=head1 NAME
+
+property - Algorithm implementation selection criteria
+
+=head1 DESCRIPTION
+
+As of OpenSSL 3.0, a new method has been introduced to decide which of multiple
+implementations of an algorithm will be used.
+The method is centered around the concept of properties.
+Each implementation has a number of properties defined and when an
+implementation is being selected, a property query is used to make the decision.
+
+=head2 Names
+
+Properties have names.
+Names fall into two categories: those reserved by the OpenSSL project
+and user defined names.
+A reserved name begins with a letter and can be followed by any number of
+letters, numbers and underscores.
+A user name is similar but B<must> include at least
+two such strings separated by periods.
+
+=head2 Properties
+
+A property is a I<name=value> pair.
+A property definition is a sequence of comma separated properties.
+There can be any number of properties in a definition.
+For example: "" defines a null property definition; "my.foo=bar" defines
+a property named I<my.foo> which has a string value I<bar> and
+"iteration.count=3" defines a property named I<iteration.count> which has a
+numeric value of I<3>.
+The full syntax for property definitions appears below.
+
+=head2 Implementations
+
+Each implementation of an algorithm can define any number of properties it has.
+For example the default provider defines the property I<default=yes> for all
+of its algorithms.
+Likewise, the FIPS provider defines I<FIPS=yes> and the legacy provider
+defines I<legacy=yes> for all of their algorithms.
+
+=head2 Queries
+
+A property query clause is a single conditional test.  For example, "FIPS=yes",
+"default!=yes" or "?iteration.count!=3".
+The first two represent mandatory clauses, such clauses B<must> match for any
+algorithm to even be in consideration.
+The third clause represents an optional clause, the more of these that match
+the better but not matching one does not rule out using an implementation.
```

Comment:
Does more matches mean it will be selected over less matches? So maybe say something about priorities if this is the case

Reply:
Ahh you say this down further..